FWCore/Services/src/InitRootHandlers.cc

```diff
@@ -438,6 +547,50 @@ namespace edm {
         edm::Exception except(edm::errors::OtherCMS, sstr.str());
         throw except;
       }
+
+        // Initialize read/write pipes for child thread.
+      int must_reload_child = parentToChild_[1];
+
+      int child_to_parent_pipe[2];
+      if (-1 == pipe2(child_to_parent_pipe, O_CLOEXEC))
+      {
+        std::ostringstream sstr;
+        sstr << "Failed to create child-to-parent pipes (errno=" << errno << "): " << strerror(errno);
+        edm::Exception except(edm::errors::OtherCMS, sstr.str());
+        throw except;
+      }
+
+      int parent_to_child_pipe[2];
+      if (-1 == pipe2(parent_to_child_pipe, O_CLOEXEC))
+      {
+        std::ostringstream sstr;
+        sstr << "Failed to create child-to-parent pipes (errno=" << errno << "): " << strerror(errno);
+        edm::Exception except(edm::errors::OtherCMS, sstr.str());
+        throw except;
+      }
+
+      close(childToParent_[0]);
+      childToParent_[0] = child_to_parent_pipe[0];
+      childToParent_[1] = child_to_parent_pipe[1];
+      parentToChild_[0] = parent_to_child_pipe[0];
+      parentToChild_[1] = parent_to_child_pipe[1];
+      if (must_reload_child >= 0)
+      {
+        int result = full_write(must_reload_child, "2");
+        if (result < 0)
+        {
+          std::ostringstream sstr;
+          sstr << "Failed to notify traceback thread to reload its file descriptors (errno=" << -result << "): " << strerror(-result);
+          edm::Exception except(edm::errors::OtherCMS, sstr.str());
+          throw except;
+        }
+      }
+      close(must_reload_child);
+
+      if (!helperThread_)
+      {
+        helperThread_.reset(new std::thread(stacktraceHelperThread));
```

Comment:
From http://en.cppreference.com/w/cpp/thread/thread/~thread we need to explicitly ```join()``` this thread at the end else termate will be called.